### PR TITLE
Update information about the `places` collection

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request updates the documented information about the `places` collection. This includes:
- Update old, dead URLs to live equivalents.
- Re-order fields to mirror the [collection schema].
- Add additional fields to match the [collection schema].

## 💭 Motivation and context ##

I was working on how this data is imported in https://github.com/cisagov/cyhy-core/pull/83 and noticed that not all fields were represented in this data dictionary. I decided to do some general cleanup when I went to add the missing fields.

## 🧪 Testing ##

Automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

[collection schema]: https://github.com/cisagov/cyhy-core/blob/9d867f753136747e709480c770cc7021c9253849/cyhy/db/database.py#L1672-L1689

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
Thanks for the cleanup! 🥇

File: NCATS_Data_Dictionary.md [hunk 1]
```diff
@@ -282,28 +282,36 @@ notification refers to
...
+- `county_fips` [string]: [FIPS
+  code](https://www.census.gov/library/reference/code-lists/ansi.html#cou)
+  for the county where this place is
+- `country` [string]: Two-letter abbreviation of the country where this
+  place is
+- `country_name` [string]: Full name of the country where this place is
+- `latitude_dms` [string]: Latitude of this place, in degrees, minutes,
+  and seconds
+- `longitude_dms` [string]: Longitude of this place, in degrees,
+  minutes, and seconds
```
Comment (jsf9k) [inline NCATS_Data_Dictionary.md]: Should we specify the format of the DMS string?  I've seen it done a few different ways, e.g., `DDDMMSS`, `DDD MM SS`.

Comment (mcdonnnj) [reply NCATS_Data_Dictionary.md]: Please let me know if https://github.com/cisagov/ncats-data-dictionary/pull/43/commits/c0b7a4c67e6ecd8fd4528a83e3e8d944ecaae365 works for you. I also added https://github.com/cisagov/ncats-data-dictionary/pull/43/commits/857a112a278a0a90d5ebe5dfc0eff6caabad92f8 as a more general catch-all for the formats of this collection's fields.

Comment (jsf9k) [reply NCATS_Data_Dictionary.md]: Perfect!

**Review by jsf9k — APPROVED** [review]

**Review by jasonodoom — APPROVED** [review]

**Pull request merged**